Gemfile

```diff
@@ -10,4 +10,5 @@ group :development do
   gem 'vterm', '>= 0.0.5' if is_unix && ENV['WITH_VTERM']
   gem 'yamatanooroti', '>= 0.0.6'
   gem "rake"
+  gem "stackprof"
```

Comment:
Done

Reply:
For example, `Net::HTTPResponse.ancestors` is equal to `[Net::HTTPResponse, Net::HTTPHeader, Object, PP::ObjectMixin, Kernel, BasicObject]`. So, [`Net::HTTPResponse#decode_content`](https://github.com/ruby/ruby/blob/91c542ad0570d02c269877226689085128038f54/lib/net/http/response.rb#L105-L107) overrides [`Net::HTTPGenericRequest#decode_content`](https://github.com/ruby/ruby/blob/91c542ad0570d02c269877226689085128038f54/lib/net/http/generic_request.rb#L59-L62), and the documents also overrides (You can check them by commands, `ri 'Net::HTTPGenericRequest#decode_content'` and `ri 'Net::HTTPResponse#decode_content'`). On the other hand, [`Net::HTTPGenericRequest#[]=`](Net::HTTPGenericRequest#[]=) overrides [`Net::HTTPHeader#[]=`](https://github.com/ruby/ruby/blob/91c542ad0570d02c269877226689085128038f54/lib/net/http/header.rb#L43-L44), but `ri 'Net::HTTPGenericRequest#[]='` shows "Implementation from HTTPHeader" because `Net::HTTPGenericRequest#[]=` has no documents. It means "the document of `Net::HTTPGenericRequest#[]=` doesn't override the document of `Net::HTTPHeader#[]=`". RDoc tracks inheritance relationships and method traversals accurately (as long as there are no bugs).

However, overriding a method complicates the story.

```ruby:a.rb
# a.rb

class A
  # Doc of method "a".
  def a
  end
end
```

```ruby:b.rb
# b.rb

require 'a'

class A
  # The second doc of method "a"
  def a
  end
end
```

```
$ rdoc -o doc_a -f ri a.rb
$ rdoc -o doc_b -f ri b.rb
$ ri -d doc_a -d doc_b 'A#a'
= A#a
(from /home/aycabta/ruby/rdoc/doc_a)
------------------------------------------------------------------------
  a()
------------------------------------------------------------------------
Doc of method "a".

(from /home/aycabta/ruby/rdoc/doc_b)
------------------------------------------------------------------------
  a()
------------------------------------------------------------------------
The second doc of method "a"
```

```ruby:c.rb
# c.rb

class A
  # Doc of method "a".
  def a
  end
end

class A
  # The second doc of method "a"
  def a
  end
end
```

```
$ rdoc -o doc_c -f ri c.rb
$ ri -d doc_c 'A#a'
= A#a
(from /home/aycabta/ruby/rdoc/doc_c)
------------------------------------------------------------------------
  a()
------------------------------------------------------------------------
Doc of method "a".
```

```
$ rdoc -o doc_ab -f ri a.rb b.rb
$ ri -d doc_ab 'A#a'
= A#a
(from /home/aycabta/ruby/rdoc/doc_ab)
------------------------------------------------------------------------
  a()
------------------------------------------------------------------------
Doc of method "a".
```

This means that "A namespace can only have one document per output directory" and "A namespace can have multiple documents if the output directory is different". So a namespace with multiple documents by the "override" in a single project can only have one document in one output directory. However, I feel this is sometimes a problem. If necessary, it might be possible to tie multiple documents in a single project to a single namespace. Please raise an issue in ruby/rdoc if you need it.

Reply:
Thank you for the clear answer, I don't think the method overriding will be a problem for our repo (https://github.com/rgeo/rgeo) but I'll keep that in mind. 

In the end we did find an issue where the doc was not propagated correctly, one colleague of mine should open a PR soon :)